miner/proposal_builder.go

```diff
@@ -499,71 +499,162 @@ func (pb *ProposalBuilder) initSharedData(ctx context.Context, current types.Lay
 	})
 }
 
-func (pb *ProposalBuilder) initSignerData(ss *signerSession, lid types.LayerID) error {
-	if ss.session.epoch != lid.GetEpoch() {
-		ss.session = session{epoch: lid.GetEpoch()}
+func (pb *ProposalBuilder) initSignerData(ctx context.Context, ss *signerSession, lid types.LayerID) error {
+	if err := pb.initSignerSessionData(ctx, &ss.session, lid, ss.signer.NodeID()); err != nil {
+		return fmt.Errorf("init signer session data: %w", err)
 	}
-	if ss.session.atx == types.EmptyATXID {
-		id, atx := pb.atxsdata.GetByEpochAndNodeID(ss.session.epoch, ss.signer.NodeID())
+	if ss.session.eligibilities.proofs == nil {
+		ss.session.eligibilities.proofs = calcEligibilityProofs(
+			ss.signer.VRFSigner(),
+			ss.session.epoch,
+			ss.session.beacon,
+			ss.session.nonce,
+			ss.session.eligibilities.slots,
+			pb.cfg.layersPerEpoch,
+		)
+		ss.log.Info("proposal eligibilities for an epoch", zap.Inline(&ss.session))
+		events.EmitEligibilities(
+			ss.signer.NodeID(),
+			ss.session.epoch,
+			ss.session.beacon,
+			ss.session.atx,
+			uint32(len(pb.shared.active.set)),
+			ss.session.eligibilities.proofs,
+		)
+	}
+	return nil
+}
+
+func (pb *ProposalBuilder) initSignerSessionData(ctx context.Context,
```

Comment:
You could remove the `ctx` - it's not used.
```suggestion
func (pb *ProposalBuilder) initSignerSessionData(
```